### ISSUE TITLE
Dropdown accessibility issues

### ISSUE THREAD

**Issue by cliffkoh** [OfficeDev/office-ui-fabric-react#5354]
Dropdown component has a bunch of issues. The most striking is that the root element of `role=listbox` has no child element with `role=option` but instead `role=listbox`. More generally, I believe we need to handle multi-select and single-select dropdowns differently.

### In the single-select dropdown case:
Fixes needed: 
- Dropdown will have `role=listbox` and the rendered element (representing selected item) will be `role=option`
- We will still render only 1 option into the DOM. But we will utilize `aria-setsize` and `aria-posinset` to let the screen reader accurately reflect what option is being rendered (this isn't done today).
- When the dropdown gets focus, we will automatically select the first option, as the spec specifies, instead of letting it stay in the placeholder text.
- We will still use `aria-haspopup="listbox"` to indicate that you can open a popup with the list of the options (not being done today)

 
### In the multi-select dropdown case
Fixes needed:
- Dropdown will not have `role=listbox`. It will instead be `aria-hasPopup="listbox"`. It will also not have `aria-activedescendant`.
  - Rationalization: In a multiple-select scenario, you cannot use the arrow keys alone to select an item. Using the arrow key always pops up the actual listbox. The field showing a string serialization of the options selected is itself not a listbox (has no listbox functionality until the popup opens).
- Aria-activedescendant will instead be on the `<FocusZone>` element which also presently has role='listbox'
- There are probably other gaps (when I was testing it, it wasn't reciting which option you're currently at, how many options were selected, etc.).
- When multiple options are selected, it is always only the first item that is read out.

**Comment by cliffkoh** [comment]
@stanleyymsft who introduced multi-select dropdown.

**Comment by cliffkoh** [comment]
Some of the issues/action items here are noted in #4676.

**Comment by betrue-final-final** [comment]
Great analysis. We might have this solved once we get a PR on this.
